gems/laa_multi_step_forms/app/controllers/steps/base_step_controller.rb

```diff
@@ -29,30 +29,36 @@ def decision_tree_class
       raise 'implement this action, in subclasses'
     end
 
-    # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
     def update_and_advance(form_class, opts = {})
       hash = permitted_params(form_class).to_h
       record = opts.fetch(:record, current_application)
       @form_object = form_class.new(
         hash.merge(application: current_application, record: record)
       )
 
+      current_application.transaction do
```

Comment:
I may be wrong here, but transactions aren't locks - a transaction simply means if one statement fails, all the rest will be rolled back. We want locking to ensure that no statements are executed while another process is also executing the statements.

Reply:
`current_application.with_lock` ?

Reply:
yes you are correct - I had it in my ActiveRecord notes that transaction does both of those things and didn't double check the official docs :/ 
that said - in this case maybe makes sense to do both as this would ensure that when a form performs multiple actions say changing the application and an associated record on save, it will roll back all changes on error

Reply:
I think locking always creates a transaction, it's just that transactions don't always create locks. c.f. https://blog.saeloun.com/2022/03/23/rails-7-adds-lock_with/

Reply:
I see! I'll remove the transaction block then

Reply:
(done)